### PR TITLE
Stop reading TargetFramework prop in props files

### PR DESCRIPTION
**Pull request by ViktorHofer**
The TargetFramework property isn't expected to be set in props files
before a project's body is evaluated.

Don't let BuildTargetFramework property fallback to TargetFramework as
BTF's sole intent is to convey the TargetFramework to filter to and not
the current selected TargetFramework. Reduce usage of BTF so that it is
only used in places where code is actually conditioned on filtering.

In addition to that, specify BuildTargetFramework as a global property for
all traversal builds so when invoking one of the traversal projects directly
(src.proj, ref.proj, etc.), BuildTargetFramework doesn't need to specified
manually to get the default behavior (filter on compatible to net6.0).
Remove the inferred `BuildingNetCoreAppVertical` because of that and
inline its meaning.

Make sendtohelix a normal NoTargets proj so that it has access to the
properties which were moved into the Directory.Build.targets file as
otherwise it wouldn't import that file.

### CODE REVIEW

**Comment by am11** [comment]
uhg, i think for Windows, we would need to escape the quote as it's already quoted. I don't have the Windows build machine, can we test it here, changing `"` to `\"` in the patch?

**Comment by ViktorHofer** [comment]
Sure

**Comment by ViktorHofer** [comment]
You mean like that?

**Comment by am11** [comment]
Yup, thanks. Lets see if it makes windows-wasm leg happy. :)
(your idea of setting it conditionally was more sensible if it doesn't work, we can try that)

**Comment by ViktorHofer** [comment]
Hmm still failing:

```
  else was unexpected at this time.
D:\workspace\_work\1\s\src\libraries\Native\build-native.proj(48,5): error MSB3073: The command ""D:\workspace\_work\1\s\src\libraries\Native\build-native.cmd" wasm Release outconfig net6.0-Browser-Release-wasm -os Browser /p:OfficialBuildId=\"\"" exited with code 255
```

**Comment by ViktorHofer** [comment]
The PR is ready for review, PTAL.

File: src/libraries/libraries-packages.proj [hunk 1]
```diff
@@ -1,7 +1,6 @@
 <Project Sdk="Microsoft.Build.Traversal" DefaultTargets="Pack">
 
   <PropertyGroup>
-    <TraversalGlobalProperties>BuildAllProjects=true</TraversalGlobalProperties>
```
Comment (Anipik) [inline src/libraries/libraries-packages.proj:4]: do we no longer need BuildAllProjects with the new version ?

**Review by Anipik — APPROVED** [review]
lgtm

File: src/libraries/libraries-packages.proj (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/libraries-packages.proj:4]: The libraries packages don't need the BuildAllProjects global property.

**Comment by ViktorHofer** [comment]
Failures are:
- https://github.com/dotnet/runtime/issues/52710
- https://github.com/dotnet/runtime/issues/11063

File: eng/pipelines/libraries/helix.yml [hunk 2]
```diff
@@ -15,7 +15,7 @@ parameters:
   scenarios: ''
 
 steps:
-  - script: $(_msbuildCommand)
+  - script: $(_msbuildCommand) -restore
```
Comment (safern) [inline eng/pipelines/libraries/helix.yml:18]: Curious, why do we need to restore `sendtohelix`?

**Pull request merged**